module-system/module-implementations/sov-ibc/src/context.rs

```diff
@@ -221,105 +366,196 @@ where
         height: Height,
         host_height: Height,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc.client_update_heights_map.set(
+            &(client_id, height),
+            &host_height,
+            *self.working_set.borrow_mut(),
+        );
+        Ok(())
     }
 
     fn store_connection(
         &mut self,
         connection_path: &ConnectionPath,
         connection_end: ConnectionEnd,
     ) -> Result<(), ContextError> {
-        todo!()
+        self.ibc.connection_end_map.set(
+            connection_path,
+            &connection_end,
+            *self.working_set.borrow_mut(),
+        );
+        Ok(())
     }
 
     fn store_connection_to_client(
         &mut self,
         client_connection_path: &ClientConnectionPath,
         conn_id: ConnectionId,
     ) -> Result<(), ContextError> {
-        todo!()
+        let mut connection_ids = self
+            .ibc
+            .connection_ids_map
+            .get(client_connection_path, *self.working_set.borrow_mut())
+            .unwrap_or_default();
+
+        connection_ids.push(conn_id);
+
+        self.ibc.connection_ids_map.set(
+            client_connection_path,
+            &connection_ids,
+            *self.working_set.borrow_mut(),
+        );
+
+        Ok(())
     }
 
     fn increase_connection_counter(&mut self) {
-        todo!()
+        let next_connection_counter = self
+            .ibc
+            .connection_counter
+            .get(*self.working_set.borrow_mut())
+            .unwrap_or_default()
```

Comment:
All counters should be initialized in `init_module()` (in `sov_ibc/src/genesis.rs`), and this should fail instead of `unwrap_or_default()`.

We should also return an `Error` in ibc-rs here

Reply:
[c97e108](https://github.com/informalsystems/sovereign-sdk/pull/1/commits/c97e108fc34ea8d98700e0f18286603257c89ef2)